### ISSUE TITLE
Feature Request: Enhance Git Graph Functionality

### ISSUE THREAD

**Issue by haseeb-heaven** [gitkraken/vscode-gitlens#3487]
The current Git graph is primarily useful for visualizing commit history, but it lacks some essential features that would significantly improve usability:

1. **Commit File Changes**: When clicking on any commit in the Git graph, it doesn't allow the user to view the files changed in that commit. This feature is essential for quickly understanding the impact of a particular commit.

2. **Switching Branches**: Currently, to view commits from different branches, I have to switch branches manually. It would be much more efficient if the Git graph allowed viewing commits across branches without the need to switch.

These features are already available in the [Git Graph extension](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph), and it would be great to see them included in GitLens, especially in the Pro version.

#### Steps to Reproduce

1. Open the Git graph in GitLens.
2. Click on any commit.
3. Note that there is no option to view files changed in the selected commit.
4. Attempt to view commits from another branch without switching the branch.

#### Expected Behavior

- Clicking on a commit should display the list of files changed in that commit.
- The ability to view commits across different branches without manually switching the branch.

#### Actual Behavior

- Currently, clicking on a commit only highlights it without providing detailed information about the files changed.
- Viewing commits from other branches requires switching the branch manually.

#### Additional Context

Implementing these features would enhance the workflow for developers using GitLens and reduce the need for switching to alternative extensions.

Thank you for considering these enhancements.

**Comment by MetRonnie** [comment]
You can double click on a commit to view the changes, but I feel like there ought to be an option to make this single click

**Comment by haseeb-heaven** [comment]
> You can double click on a commit to view the changes, but I feel like there ought to be an option to make this single click

Yes i saw that but still its pro version paid and Git graph is free offers better views of Git branches and also has features of switching to other branches without needing to checkout
